### PR TITLE
Add inline documentation for `getMaxTableNameLength` function

### PR DESCRIPTION
**Pull request by yariks5s**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 43586f14f1f374c28de8cc7257599cdf5d952455 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/73402/43586f14f1f374c28de8cc7257599cdf5d952455/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Functions/getMaxTableNameLength.cpp [hunk 1]
```diff
@@ -92,7 +92,37 @@ class FunctionGetMaxTableNameLength : public IFunction, WithContext
 
 REGISTER_FUNCTION(getMaxTableName)
 {
-    factory.registerFunction<FunctionGetMaxTableNameLength>();
+    factory.registerFunction<FunctionGetMaxTableNameLength>(FunctionDocumentation{
```
Comment (rschu1ze) [inline src/Functions/getMaxTableNameLength.cpp:95]: Side note: Suggest to also rename the class name `FunctionGetMaxTableNameLength` to `FunctionGetMaxTableNameLengthForDatabase`.

File: src/Functions/getMaxTableNameLength.cpp [hunk 2]
```diff
@@ -92,7 +92,37 @@ class FunctionGetMaxTableNameLength : public IFunction, WithContext
...
+        .description=R"(
+Returns the maximum table name length in a specified database.
+
+Syntax:
```
Comment (rschu1ze) [inline src/Functions/getMaxTableNameLength.cpp:99]: That's not how the syntax, arguments, returned value etc. are supposed to be described as inline-docs. Please see src/Common/FunctionDocumentation.h for the proper way (basically populate different member variables).

Comment (yariks5s) [reply src/Functions/getMaxTableNameLength.cpp]: done

File: src/Functions/getMaxTableNameLength.cpp (hunk 1 above)
Comment (yariks5s) [reply src/Functions/getMaxTableNameLength.cpp]: done

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**